src/context/PokemonsProvider.js

```diff
@@ -0,0 +1,20 @@
+import { useState, useContext, createContext, useMemo } from 'react';
+
+export const PokemonsContext = createContext();
+
+export default function PokemonsProvider( { children } ) {
+	const [ pokemons, setPokemons ] = useState( [] );
+	const [ currentPokemonId, setCurrentPokemonId ] = useState( -1 );
+
+	const currentPokemon = useMemo( () => pokemons[ currentPokemonId ], [ pokemons, currentPokemonId ] );
```

Comment:
The useMemo is useless - each time that one of the states changes you will definitely need to re-calculate the current pokemon anyway..

Reply:
Removed